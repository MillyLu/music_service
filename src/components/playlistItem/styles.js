import styled from "styled-components";

import {ReactComponent as NoteSvg} from '../svg/note/note.svg';
import {ReactComponent as LikeSvg} from '../svg/like/like.svg'

export const A = styled.a`
font-style: normal;
font-weight: 400;
font-size: 16px;
line-height: 24px;
color: #FFFFFF;
&:visited {
    text-decoration: none;
    font-family: 'StratosSkyeng', sans-serif;
    cursor: pointer;
}
`;

export const PlaylistItem = styled.div`
width: 100%;
display: block;
margin-bottom: 12px;
`;

export const PlaylistTrack = styled.div`
display: flex;
flex-direction: row;
justify-content: space-between;
align-items: center;
`;

export const TrackTitle = styled.div`
display: flex;
flex-direction: row;
align-items: center;
max-width: 447px;
`;

export const TrackTitleImage = styled.div`
width: 51px;
height: 51px; 
padding: 16px;
background: #313131;
display: flex;
align-items: center;
justify-content: center;
margin-right: 17px;
`;

export const TrackTitleSvg = styled(NoteSvg)`
width: 18px;
height: 17px;
fill: transparent;
stroke: #4E4E4E;

`;

export const TrackTitleLink = styled(A)`
font-style: normal;
font-weight: 400;
font-size: 16px;
line-height: 24px;
color: #FFFFFF;
&:visited {
    text-decoration: none;
    font-family: 'StratosSkyeng', sans-serif;
    cursor: pointer;
}
`;

export const TrackTitleSpan = styled.span`
font-style: normal;
font-weight: 400;
font-size: 16px;
line-height: 24px;
color: #4E4E4E;
`;

export const TrackAuthor = styled.div`
width: 200px;
display: flex;
justify-content: flex-start;
`;

export const TrackAuthorLink = styled(A)`
font-style: normal;
font-weight: 400;
font-size: 16px;
line-height: 24px;
color: #FFFFFF;
text-align: left;
`;

export const TrackAlbum = styled.div`
width: 150px;
`;

export const TrackAlbumLink = styled(A)`
font-style: normal;
font-weight: 400;
font-size: 16px;
line-height: 24px;
color: #696969;
`;

// eslint-disable-next-line no-undef
export const TrackTimeSvg = styled(LikeSvg)`
width: 14px;
height: 12px;
margin-right: 17px;
fill: transparent;
stroke: #696969;

&:hover, &:focus  {
    stroke: ${props => props.theme === "light" ? "#AD61FF" :  "#ACACAC"}; 
    scale: 1.05;
    cursor: pointer;
  }
&:active  {
    fill: ${props => props.theme === "light" ? "#AD61FF" :  "#696969"};
    stroke: ${props => props.theme === "light" ? "#AD61FF" :  "#ACACAC"};
    cursor: pointer;
  }
`;
// !!!!!!!!!!!!!!!!!!!!!!!//
export const TrackTitleText = styled.div`
width: 350px;
`;

export const TrackTime = styled.div`
width: 60px;
display: flex;
flex-direction: raw;
align-items: center;
`

export const TrackTimeText = styled.span`
font-style: normal;
font-weight: 400;
font-size: 16px;
line-height: 24px;
text-align: right;
color: #696969;
`;
