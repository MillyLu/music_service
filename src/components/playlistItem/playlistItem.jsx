/* eslint-disable no-unneeded-ternary */
import { useContext} from 'react';
import * as Styled from './styles';
import { ThemeContext } from '../../ThemeProvider';

export function PlayListItem({id, author, album, time, title, subtitle, item, handleChangeOfFavorites, onFetchTrack}) { 
    const {theme} = useContext(ThemeContext);


    return(
        <Styled.PlaylistItem onClick={() => onFetchTrack(id)}> 
            <Styled.PlaylistTrack>
                <Styled.TrackTitle>
                    <Styled.TrackTitleImage style={{ backgroundColor: theme === "light" ? "#F6F4F4" : "#313131" }}>
                        <Styled.TrackTitleSvg alt='music' >
                            
                        </Styled.TrackTitleSvg>
                    </Styled.TrackTitleImage>
                    <Styled.TrackTitleText>
                        <Styled.TrackTitleLink onClick={() => onFetchTrack(id)} href='#' style={{ color: theme === "light" ? "#000000" : "#FFFFFF" }}>{title}<Styled.TrackTitleSpan>{subtitle}</Styled.TrackTitleSpan></Styled.TrackTitleLink>
                       
                    </Styled.TrackTitleText>
                </Styled.TrackTitle>
                <Styled.TrackAuthor>
                    <Styled.TrackAuthorLink href='#' style={{ color: theme === "light" ? "#000000" : "#FFFFFF" }}>{author}</Styled.TrackAuthorLink>
                </Styled.TrackAuthor>
                <Styled.TrackAlbum>
                    <Styled.TrackAlbumLink href='#'>{album}</Styled.TrackAlbumLink>
                </Styled.TrackAlbum>
                <Styled.TrackTime>
                    <Styled.TrackTimeSvg  theme={theme} alt='time' onClick={(event) => {
          handleChangeOfFavorites(item)
          event.stopPropagation()
        }}>
                    </Styled.TrackTimeSvg>
                    <Styled.TrackTimeText>{time}</Styled.TrackTimeText> 
                </Styled.TrackTime>
            </Styled.PlaylistTrack>
        </Styled.PlaylistItem>
    )



}
