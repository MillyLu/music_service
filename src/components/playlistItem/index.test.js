/* eslint-disable no-undef */
import * as ShallowRenderer from 'react-shallow-renderer';
// import { ShallowRenderer, createRenderer } from 'react-shallow-renderer';
import { PlayListItem } from './playlistItem';



describe('<PlayListItem/>', () => {
        it('should render successfully and match snapshot', () => {
            const shallowRenderer = ShallowRenderer.createRenderer();
            const renderer = shallowRenderer;
            renderer.render(<PlayListItem />);
            const result = renderer.getRenderOutput();
            expect(result).toMatchSnapshot();
          })
      });