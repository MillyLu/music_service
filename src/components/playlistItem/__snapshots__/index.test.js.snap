// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlayListItem/> should render successfully and match snapshot 1`] = `
<styled.div
  onClick={[Function]}
>
  <styled.div>
    <styled.div>
      <styled.div
        style={
          Object {
            "backgroundColor": "#313131",
          }
        }
      >
        <Styled(Component)
          alt="music"
        />
      </styled.div>
      <styled.div>
        <Styled(styled.a)
          href="#"
          onClick={[Function]}
          style={
            Object {
              "color": "#FFFFFF",
            }
          }
        >
          <styled.span />
        </Styled(styled.a)>
      </styled.div>
    </styled.div>
    <styled.div>
      <Styled(styled.a)
        href="#"
        style={
          Object {
            "color": "#FFFFFF",
          }
        }
      />
    </styled.div>
    <styled.div>
      <Styled(styled.a)
        href="#"
      />
    </styled.div>
    <styled.div>
      <Styled(Component)
        alt="time"
        onClick={[Function]}
        theme={0}
      />
      <styled.span />
    </styled.div>
  </styled.div>
</styled.div>
`;
